service/src/main/java/org/cbioportal/genome_nexus/service/internal/PdbDataServiceImpl.java

```diff
@@ -25,75 +19,35 @@
 @Service
 public class PdbDataServiceImpl implements PdbDataService
 {
-    private static final Log LOG = LogFactory.getLog(PdbDataServiceImpl.class);
-
-    @Autowired
-    private SimpleCacheRepository cacheRepository;
-
     @Autowired
-    private PdbHeaderDataFetcher externalResourceFetcher;
+    private CachedPdbHeaderFetcher cachedExternalResourceFetcher;
 
     @Override
-    public PdbHeader getPdbHeader(String pdbId) throws PdbHeaderNotFoundException, PdbHeaderWebServiceException {
-        boolean saveRawValue = true;
+    public PdbHeader getPdbHeader(String pdbId) throws PdbHeaderNotFoundException, PdbHeaderWebServiceException
+    {
         PdbHeader pdbHeader = null;
-        String rawValue = null;
 
-        // try to get the data from database first
         try {
-            SimpleCacheEntity entity = cacheRepository.findOne(pdbId);
+            // get the PDB data from the web service and save it to the DB
+            pdbHeader = this.cachedExternalResourceFetcher.fetchAndCache(pdbId);
 
-            if (entity != null)
-            {
-                // we have the information in the cache already!
-                rawValue = entity.getValue();
+            // include original pdb id value too
+            if (pdbHeader != null) {
+                pdbHeader.setPdbId(pdbId);
```

Comment:
Actually, this is already done by the PDB file parser. But leaving this code segment just in case the value is missing/different in the file.